src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -30,6 +31,19 @@ internal class SilentRequest : RequestBase
             _silentParameters = silentParameters;
         }
 
+        private bool _canExecuteBroker
```

Comment:
We know `PreRunAsync()` will always be called before `ExecuteAsync`, so I would set `private bool _canExecuteBrokerSilentRequest = false` by default, and then in `PreRunAsync()`, if it's true, the value will change, then just check the value of `_canExecuteBrokerRequest` in `ExecuteAsync()`...what do you think?
```csharp
  private void _IsBrokerAvailableForSilentRequest
        {
                if (!_canExecuteBrokerSilentRequest)
                {
                   _canExecuteBrokerSilentRequest = AuthenticationRequestParameters.IsBrokerConfigured && ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth();
                }         
        }
```

Reply:
Are there any tests that can be done for this using the IBroker? Like, it should go through a normal flow and not try to use android, so we can at least test for that, correct? I'm not too familiar w/the android implementation.

---
In reply to: [419826655](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419826655) [](ancestors = 419826655)

Reply:
Agreed with testing comment, every line in SilentAuth needs to be unit tested.